tests/common/platform/transceiver_utils.py

```diff
@@ -43,10 +43,23 @@ def all_transceivers_detected(dut, interfaces):
     Check if transceiver information of all the specified interfaces have been detected.
     """
     db_output = dut.command("redis-cli --raw -n 6 keys TRANSCEIVER_INFO\*")["stdout_lines"]
-    not_detected_interfaces = [intf for intf in interfaces if "TRANSCEIVER_INFO|%s" % intf not in db_output]
-    if len(not_detected_interfaces) > 0:
-        logging.info("Interfaces not detected: %s" % str(not_detected_interfaces))
-        return False
+
+    for intf in interfaces:
+        if dut.facts['hwsku'] == 'Nokia-7215':
```

Comment:
we should not hardcode hwsku here.

Reply:
The reason was for hwsku based check is to not affect other platform that may have 1g sfp or copper sfp , by removing hwsku will introduce unintentional skip for 1g sfp

Reply:
@lguohan can you please review based on my comment above

Reply:
@rawal01 Can you elaborate more why Nokia 7215 needs to skip 1GP ports in transceiver test? Are these ports RJ-45 ports?

Reply:
@yxieca Yes these are RJ-45 ports

Reply:
@jleveque how do you want to deal with RJ-45 ports in transceiver tests? I think we have other M0 device that will have the same issue if we run transceiver test. It is better to have a more generic solution to find and filter out RJ-45 (or non-sfp) ports in the test. Are we able to use ethertool to figure that out and filter out these ports? if that is not acceptable, maybe pass in a list of ports to ignore (or some other conditions, e.g speed=1000) from the actual test. I think we want to keep helper functions generic.

Reply:
Maybe we could utilize a data file like hwsku.json?

Reply:
@jleveque 
I looked at existing hwsku.json file for some of the platforms it exists and I can see   "default_brkout_mode" as an field present, we can add a new field "port_type": with possible values "SFP" and "Non-sfp" or a different name if you already have something in your mind ?

Reply:
@rawal01: Yes, something like that is what I envisioned. Maybe rather than `Non-SFP` we could specify the actual port type, i.e., `RJ45`. So for now we would expect either `SFP` or `RJ45`. But in the future, we could extend this if we see a need to differentiate the different SFP variants (e.g., `RJ45, SFP, SFP+, SFP28, QSFP+, QSFP28`).

Reply:
@jleveque 
Thank you for your quick response, it makes sense port_type RJ45 and sfp+ to differentiate fixed ports vs pluggable for now and have exception in transceiver_utils.py  for RJ45 ports if hwsku.json specifies such ports, as you said more variants can be added as required for other purpose.